src/doc_builder/external.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import tempfile
+import zlib
+
+import git
+import requests
+
+
+HF_DOC_PREFIX = "https://huggingface.co/docs/"
+EXTERNAL_DOC_OBJECTS_CACHE = {}
+HUGGINFACE_LIBS = [
+    "accelerate",
+    "datasets",
+    "evaluate",
+    "huggingface_hub",
+    "optimum",
+    "tokenizers",
+    "transformers",
+]
+
+
+def post_process_objects_inv(object_data, doc_url):
+    """
+    Post-processes the data in sphinx-like format to get a dictionary object_name: link in doc.
+
+    Args:
+        object_data (`str`): The data in the `objects.inv` object (except the first 4 lines).
+        doc_url (`str`): The documentation url of the package.
+    """
+    links = {}
+    for line in object_data:
+        if len(line) == 0:
+            continue
+        name, typ, _, link = line.split(" ")[:4]
+        if typ in ["py:class", "py:function", "py:method"]:
+            link = link.replace("$", name)
+            links[name] = f"{doc_url}/{link}"
+    return links
+
+
+def get_stable_version(package_name):
+    """
+    Gets the version of the last release of a package.
+
+    Args:
+        package_name (`str`): The name of the package.
+    """
+    github_url = f"https://github.com/huggingface/{package_name}"
+    try:
+        # Get the version tags from the GitHub repo in decreasing order (that's what '-v:refname' means)
+        result = git.cmd.Git().ls_remote(github_url, sort="-v:refname", tags=True)
+    except git.GitCommandError:
+        return "main"
```

Comment:
Nope, the redirect works properly to download the objects.inv and point to all the links (plus they should really switch to main) (test is using Datasets on purpose since it has some many specific quirks!)